Sofa/Component/Constraint/Lagrangian/Solver/src/sofa/component/constraint/lagrangian/solver/GenericConstraintSolver.cpp

```diff
@@ -1119,6 +1151,222 @@ void GenericConstraintProblem::unbuiltGaussSeidel(SReal timeout, GenericConstrai
     }
 }
 
+void GenericConstraintProblem::NNCG(GenericConstraintSolver* solver, int iterationNewton)
+{
+    if(!solver)
+        return;
+    int dimension = getDimension();
+    int maxIterations = iterationNewton;
+
+    if(!dimension)
+    {
+        currentError = 0.0;
+        currentIterations = 0;
+        return;
+    }
+
+
+    lam.clear();
+    lam.resize(dimension);
+    deltaF.clear();
+    deltaF.resize(dimension);
+    deltaF_new.clear();
+    deltaF_new.resize(dimension);
+    p.clear();
+    p.resize(dimension);
+
+
+    double *dfree = getDfree();
```

Comment:
Pay attention to the floating point types. Here `double` must be `SReal`. It is probably the case in the rest of the code